native/src/modules/common/openExternalUrl.js

```diff
@@ -0,0 +1,19 @@
+// @flow
+
+import { Linking } from 'react-native'
+import { InAppBrowser } from 'react-native-inappbrowser-reborn'
+import buildConfig from '../app/constants/buildConfig'
+
+const openExternalUrl = async (url: string) => {
+  try {
+    if (await InAppBrowser.isAvailable()) {
+      await InAppBrowser.open(url, {
+        toolbarColor: buildConfig().lightTheme.colors.themeColor
+      })
+    } else { Linking.openURL(url) }
```

Comment:
Should we check here whether we can open it like in tunews?
```
    const supported = await Linking.canOpenURL(tunewsWebsiteUrl)
    if (supported) {
      await Linking.openURL(tunewsWebsiteUrl)
    }
```

Reply:
Sure, we could. But what happens if it is now supported? Imo better to just open it and the user gets an error there.

Reply:
I suppose that Linking.open will crash the app :D But I'm not sure what the behavior is.

We could show a snackbar when an error occurrs.

Reply:
Oh oke then the catch block should hit.
I think a good implementation would be to check `canOpenURL()`. If the result is false then show "You do not have an app installed to open this link" in a snackbar.

If an error occurs when calling `canOpenURL` or `openLink` then show some other snackbar.

Do we already have a snackbar for this usecase? If yes then I would say that we use it here, if not then we can create a separate task.

Reply:
We don't have a snackbar yet for such a usecase yet, nope. Will only be implemented later.

Reply:
I don't think this is not relevant at all. The only case this can occur is if the user does not have custom tabs support (or the pendant on iOS which is probably never the case because of apple) AND no browser is installed. Or is there any other usecase where this will happen?

Reply:
For example if somebody in the backend embeds an email link, or geo link and there is no app.
This is just proper/better error handling. Not much benefit, but also not much work.

Reply:
Okay, so just wrapping in canOpenURL and adding a todo for the snackbar?

Reply:
I think that would be good!